Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1394,91 +1392,94 @@ internal BlockLocator GetBlockLocator(int threshold = 10)
                 topmostCommon.Index,
                 topmostCommon
             );
-
-            bool reorged = !Tip.Equals(topmostCommon);
-            if (renderBlocks && reorged)
+            try
             {
-                foreach (IRenderer<T> renderer in Renderers)
+                _rwlock.EnterWriteLock();
```

Comment:
Could we use a separated lock from _rwlock (i.e. _renderLock)?

Reply:
Unfortunately, it doesn't seem to work as expected when only the render is locked. 😢

Reply:
🆗